### DISCUSSION TITLE
Implementation of a per-region transformation matrix

### DISCUSSION THREAD

**Discussion by abiswas97** [visgl/deck.gl#8489] [Q&A]
Hi, I'm looking for some guidance on a specific use case. I'm working with a variation of MVT layers, where I have a pyramid of custom vector tiles that are rendered onto a base Tile Layer. 

We want to enable registration of these vectors to a base cartesian image, so multiple transformation matrices would be applied to different regions in the viewport. For eg. fixing registration in just one corner of the view. The goal behind this approach is to allow for 'editable' vector tiles, where we store the transformation matrix of a region instead of altering the underlying tiled data. 

My question is - 
What would be an efficient approach to pass per region transform matrices to the layer instance? From what I understand, the `modelMatrix` prop applies a transform to the entire layer, and wouldn't be suitable here. Would I want to directly use math.gl on my data array and pass the transformed data array directly? From the math.gl docs, I see that it doesn't inherently make use of the gpu - what would be needed to allow for the data transform to make use of the GPU?

Apologies for a poorly formed question - quite new to this field and would be grateful for any guidance :))

**Comment by Pessimistress** [comment]
The TileLayer renders one or more layers for each tile. Each of these sublayers can have their own `modelMatrix`. In fact, that's how MVTLayer is implemented.

**Reply by abiswas97** [reply]
Didn't realise this - I'll take a look! 

Off the top of my head though, the solution I'm trying to go for is to conditionally apply transformations to regions of data. Specifically, my tiles render out as geojson layers within TileLayer. I'm looking to do something like "apply transformation T1 to all features within only the bottom corner of the image, and apply T2 to a 5k x 5k (pixels) box in the centre). 

The naive solution seems to be to preprocess all data before load, and run a processing step on affected data indices when a new transformation is applied. But I fear this could become slow when done outside of the gpu context. 

How would I go about ensuring that I apply custom transformations to a portion of the data, but in the gpu context?
